src/coreclr/vm/tieredcompilation.cpp

```diff
@@ -1016,9 +1088,24 @@ CORJIT_FLAGS TieredCompilationManager::GetJitFlags(PrepareCodeConfig *config)
         NativeCodeVersion::OptimizationTier newOptimizationTier;
         if (!methodDesc->RequestedAggressiveOptimization())
         {
+            NativeCodeVersion::OptimizationTier currentTier = nativeCodeVersion.GetOptimizationTier();
+            if (currentTier == NativeCodeVersion::OptimizationTier::OptimizationTierInstrumented)
+            {
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_BBINSTR);
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_TIER0);
+                return flags;
+            }
```

Comment:
If `QuickJit` is disabled, it looks like this would still generate instrumented tier 0 code. Would it make sense to move this block inside the `if (g_pConfig->TieredCompilation_QuickJit())` block below?